C/include/c4Socket.h

```diff
@@ -42,7 +42,9 @@ extern "C" {
         kWebSocketCloseCantFulfill      = 1011, // Can't fulfill request due to "unexpected condition"
         kWebSocketCloseTLSFailure       = 1015, // Never sent, only received
 
-        kWebSocketCloseFirstAvailable   = 4000, // First unregistered code for freeform use
+        kWebSocketCloseUserTransient    = 4001, // any transient message endpoint websocket closure by user
+        kWebSocketCloseUserPermanent    = 4002, // any permanent message endpoint websocket closure by user
+        kWebSocketCloseFirstAvailable   = 4003, // First unregistered code for freeform use
```

Comment:
Bump this up to 5000 so it won't be affected if we have to add more errors above it. (Also check that none of the platforms are using codes in the 4000 range.)